### PR TITLE
Feature/tasks

### PR DESCRIPTION
**Pull request by vonElfvin**
# Jira Issue: Load Balancing

## What?
- MVP of Tasks
- Ability to reject a task
- Ability to move a task
- (Hopefully) Snappier fetching of tasks upon checkin+checkout
- Ability to "spy" on other users
- Remove lock on engaging with tasks when not checked in
- Have browser title say the number of tasks assigned to you and tasks in total
- Highlight unassigned tasks in a separate tab
- Ability to select which market you work in and if you do Questions and/or Claims

## Why?
- Get load balancing MVP out so we can start fixing issues

## Optional screenshots
Checked in with tasks assigned to your queue:
image

Watching another user's queue:
image

## Optional checklist
- [ ] Updated `src/changelog.ts`
- [x] Codescouted
- [ ] Unit tests written
- [x] Tested locally

### CODE REVIEW

**Review by falklinus — APPROVED** [review]
📗 ✅

File: src/portals/hope/features/tasks/hooks/use-reject-task.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  mutation RejectTask($taskId: ID!) {
+    task_reject(taskId: $taskId) {
+      id
+      assignedTo
+    }
+  }
+`
+
+export const useRejectTask = () => {
+  const [resolveTask, { loading }] = useRejectTaskMutation()
```
Comment (falklinus) [inline src/portals/hope/features/tasks/hooks/use-reject-task.ts:19]:
```suggestion
  const [rejectTask, { loading }] = useRejectTaskMutation()
```

File: src/portals/hope/features/tasks/hooks/use-reject-task.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  const reject = (taskId: string) =>
+    resolveTask({
```
Comment (falklinus) [inline src/portals/hope/features/tasks/hooks/use-reject-task.ts:22]:
```suggestion
    rejectTask({
```

File: src/portals/hope/features/tasks/hooks/use-tasks.ts [hunk 3]
```diff
@@ -17,65 +120,159 @@ interface UseTasksVariables {
...
+    if (!selectedTask) {
+      return
+    }
+
+    toast.promise(
+      resolve(taskId).then(() => onResolve?.(taskId)),
+      {
+        success: 'Resolved',
+        loading: 'Resolving',
+        error: 'Could not resolve',
```
Comment (falklinus) [inline src/portals/hope/features/tasks/hooks/use-tasks.ts:258]: No message from back-office?

**Review by falklinus — APPROVED** [review]
Really nice 👍  Added some small comments

**Comment by vonElfvin** [comment]
Move 👇 
image

File: shared/@hedvig-ui/utils/text.ts [hunk 4]
```diff
@@ -29,3 +29,17 @@ export const convertCamelcaseToTitle = (text: string) =>
...
   /^-?\d+$/.test(s) || /^\d+\.\d+$/.test(s)
+
+export const convertEmailToName = (text: string) =>
+  text
+    .split('@')[0]
+    .split('.')
+    .map((word) => {
+      return word.charAt(0).toUpperCase() + word.slice(1)
+    })
+    .join(' ')
```
Comment (rasmusguterstam) [inline shared/@hedvig-ui/utils/text.ts:40]: Isn't this already on the `User` type? I think this is the default value set for name in back-office!

File: src/portals/hope/features/tasks/TaskChat.tsx [hunk 5]
```diff
@@ -99,25 +119,64 @@ export const TaskChat: React.FC<{
...
+                {checkedInUsers.map((assignedTo) => (
+                  <DropdownOption
+                    key={assignedTo}
+                    onClick={() => {
+                      onMove(task.id, assignedTo)
+                    }}
+                    selected={assignedTo === task.assignedTo}
+                  >
+                    <MoveDropdownText selected={assignedTo === task.assignedTo}>
+                      {convertEmailToName(assignedTo)}
```
Comment (rasmusguterstam) [inline src/portals/hope/features/tasks/TaskChat.tsx:135]: Ah alright, got it

File: src/portals/hope/features/tasks/hooks/use-reject-task.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import gql from 'graphql-tag'
+import {
+  AllTasksDocument,
+  useRejectTaskMutation,
+} from 'types/generated/graphql'
+
+gql`
```
Comment (rasmusguterstam) [inline src/portals/hope/features/tasks/hooks/use-reject-task.ts:7]: Really nice 🤝

File: src/portals/hope/features/tasks/list-items/TaskListItem.tsx [hunk 7]
```diff
@@ -172,16 +213,25 @@ export const TaskListItem: React.FC<
...
       >
+        <UserCircle
+          assignedToMe={assignedToMe}
+          unassigned={task.assignedTo === null}
+        >
+          {task.assignedTo
+            ?.split('@')[0]
+            .split('.')
+            .map((word) => word.charAt(0).toUpperCase())
+            .join('')}
```
Comment (rasmusguterstam) [inline src/portals/hope/features/tasks/list-items/TaskListItem.tsx:227]: The email helper instead?

**Review by rasmusguterstam — APPROVED** [review]
Sending my prayers for this deployment 🙏

File: shared/@hedvig-ui/utils/text.ts (hunk 4 above)
Comment (vonElfvin) [reply shared/@hedvig-ui/utils/text.ts:40]: Yes it is on the user but my lazy ass don't want to fetch the user here

**Pull request merged**